src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,109 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+interface Props {
+  listOfTodos : Todo[],
+  callbackForUserSelect: (id : number) => void;
+}
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+export const TodoList: React.FC<Props> = (
+  { listOfTodos, callbackForUserSelect },
+) => {
+  const [currentFilter, setCurrentFilter] = useState('');
+  const [filterBySelect, setFilterBySelect] = useState('all');
 
-          <button
-            className="
+  function sorter(list : Todo[]) {
+    switch (filterBySelect) {
+      case 'completed': {
+        return list.filter(el => el.completed === true);
+      }
+
+      case 'active': {
+        return list.filter(el => el.completed === false);
+      }
+
+      default: {
+        return list;
+      }
+    }
+  }
+
+  return (
+    <div className="TodoList">
+      <h2>Filter todos:</h2>
+      <label>
+        <p>filter</p>
+        <input
+          type="text"
+          onChange={(event) => {
+            setCurrentFilter(event.target.value);
+          }}
+        />
+      </label>
+      <label>
+        <p>
+          Status
+        </p>
+        <select
+          onChange={(event) => {
+            setFilterBySelect(event.target.value);
+          }}
+        >
+          <option value="all">
+            all
+          </option>
+          <option value="completed">
+            completed
+          </option>
+          <option value="active">
+            active
+          </option>
+        </select>
+      </label>
+
+      <h2>Todos:</h2>
+
+      <div className="TodoList__list-container">
+        <ul className="TodoList__list">
+          {sorter(listOfTodos).map(singleTodo => (
+            singleTodo.title.includes(currentFilter) && (
```

Comment:
You shouldn't check it each time for each todo
Call filter with needed callback and store the result in some variable before the return statement